rewrite/rewrite/python/printer.py

```diff
@@ -0,0 +1,1350 @@
+                                          PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, ",", ")", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.DOUBLE_STAR:
+            self._visit_container_printer("**", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, "", "", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.KEY_VALUE:
+            self._visit_container_printer("", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, ":", "", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.KEYWORD:
+            self._visit_container_printer("", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, "=", "", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.MAPPING:
+            self._visit_container_printer("{", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, ",", "}", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.OR:
+            self._visit_container_printer("", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, "|", "", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.SEQUENCE:
+            self._visit_container_printer("", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, ",", "", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.SEQUENCE_LIST:
+            self._visit_container_printer("[", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, ",", "]", p)
+        elif pattern.kind in {MatchCase.Pattern.Kind.GROUP, MatchCase.Pattern.Kind.SEQUENCE_TUPLE}:
+            self._visit_container_printer("(", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, ",", ")", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.STAR:
+            self._visit_container_printer("*", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, "", "", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.VALUE:
+            self._visit_container_printer("", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, "", "", p)
+        elif pattern.kind == MatchCase.Pattern.Kind.WILDCARD:
+            self._visit_container_printer("_", children, PyContainer.Location.MATCH_CASE_PATTERN_CHILDREN, "", "", p)
+
+        return pattern
+
+    def visit_slice(self, slice_: Slice, p: PrintOutputCapture[P]) -> J:
+        self.before_syntax(slice_, PySpace.Location.SLICE_PREFIX, p)
+        self.visit_right_padded(slice_.padding.start, PyRightPadded.Location.SLICE_START, p)
+        p.append(':')
+        if slice_.padding.stop is not None:
+            self.visit_right_padded(slice_.padding.stop, PyRightPadded.Location.SLICE_STOP, p)
+            if slice_.padding.step is not None:
+                p.append(':')
+                self.visit_right_padded(slice_.padding.step, PyRightPadded.Location.SLICE_STEP, p)
+        return slice_
+
+    # CUSTOM VISIT METHODS FOR VISIT CONTAINER, VISIT SPACE, VISIT RIGHT PADDED, VISIT LEFT PADDED
+    def _visit_container_printer(self, before: str, container: Optional[JContainer[J2]],
```

Comment:
I think I would rather rename these overloads to e.g. `_print_container` to keep the code a bit more readable.